application/common/dos.py

```diff
@@ -102,21 +104,24 @@ def is_valid(self) -> bool:
         return None not in (self.easting, self.northing, self.latitude, self.longitude)
 
 
-def get_matching_dos_services(odscode: str) -> List[DoSService]:
+def get_matching_dos_services(odscode: str, org_type_id: str) -> List[DoSService]:
     """Retrieves DoS Services from DoS database
 
     Args:
         odscode (str): ODScode to match on
+        org_type_id (str): OrganisationType to match on
 
     Returns:
         list[DoSService]: List of DoSService objects with matching first 5
         digits of odscode, taken from DoS database
     """
-
-    logger.info(f"Searching for DoS services with ODSCode that matches first 5 digits of '{odscode}'")
-
-    sql_query = f"SELECT {', '.join(DoSService.db_columns)} FROM services WHERE odscode LIKE %(ODS_5)s"
-    named_args = {"ODS_5": f"{odscode[0:5]}%"}
+    odscode = get_new_odscode_for_dos(odscode, org_type_id)
+    logger.info(f"Searching for '{org_type_id}' DoS services with ODSCode that matches '{odscode}'")
+    sql_query = (
+        f"SELECT {build_select_statement()} FROM services s, servicetypes st"
+        " WHERE s.typeid = st.id and odscode LIKE %(ODS)s"
```

Comment:
I'm surprised there isn't a join on this table for the service type id to service type name. Has this part been tested?

Reply:
I tested  yesterday and working fine with current deployed version

Reply:
The DoSService object doesn't currently support service type name yet, I was going to add it to a branch in the future but I don't see any changes to it so I dont think collecting the data in the db query will do anything at the moment.